arch/xtensa/core/elf.c

```diff
@@ -174,3 +175,106 @@ int arch_elf_relocate_global(struct llext_loader *ldr, struct llext *ext, const
 	return xtensa_elf_relocate(ldr, ext, rel, (uintptr_t)link_addr, rel_addr, type,
 				   ELF_ST_BIND(sym->st_info), NULL);
 }
+
+uintptr_t arch_d2i_address(uintptr_t ptr) {
+	if(IN_RANGE(ptr, SRAM1_DRAM_START, SRAM1_DRAM_START + SRAM1_SIZE)) {
+		return ptr + IRAM_DRAM_OFFSET;
+	}
+	return 0;
+}
+
+uintptr_t arch_i2d_address(uintptr_t ptr) {
+	if(IN_RANGE(ptr, SRAM1_IRAM_START, SRAM1_IRAM_START + SRAM1_SIZE)) {
+		return ptr - IRAM_DRAM_OFFSET;
+	}
+	return 0;
+}
+
+uintptr_t arch_address_mangler(uintptr_t ptr, unsigned char sym_type,
+					elf32_word sh_type) {
+	if(sym_type == STT_FUNC || sh_type == SHT_INIT_ARRAY ||
```

Comment:
Ensure there's a space between any keyword and its `(`.
```suggestion
	if (sym_type == STT_FUNC || sh_type == SHT_INIT_ARRAY ||
```